bluetooth:fix zblue multi-controller build break

chengkai15 opened the pull request:
bug: v/57425

depends-on: [ open-vela/external_zblue/pull/48]

## Summary

fix zblue multi-controller build break

## Impact

multi-controller 

## Testing

chengkai15 committed:
bluetooth:fix zblue multi-controller build break

bug: v/57425

Signed-off-by: chengkai <chengkai@xiaomi.com>

gzh-terry commented on service/stacks/zephyr/sal_adapter_interface.c:
```diff
@@ -488,7 +489,6 @@ static void zblue_on_discovery_complete_cb(const struct bt_br_discovery_result*
     size_t count)
 {
     adapter_on_discovery_state_changed(BT_DISCOVERY_STOPPED);
-    return;
```
Why this is necessary for a void function?

chengkai15 replied:
so remove  it

chengkai15 committed:
bluetooth: fix multi-contoller br enable fail

bug: v/57425

rootcause: bt_conn_cb_register_mc would refer null conn_ctx when do
conn init before enable.

Signed-off-by: chengkai <chengkai@xiaomi.com>

chengkai15 committed:
bluetooth: fix multi-contoller ble enable fail

bug: v/57425

Signed-off-by: chengkai <chengkai@xiaomi.com>

JinxianChe commented on service/stacks/zephyr/sal_adapter_interface.c:
```diff
@@ -399,10 +399,11 @@ static void zblue_on_bond_deleted(uint8_t id, const bt_addr_le_t* peer)
 {
     uint8_t state = BT_BREDR_STACK_STATE_OFF;
 
+    UNUSED(dev_id);
```
Are interfaces for multiple controllers only provided now?

chengkai15 replied:
> Are interfaces for multiple controllers only provided no

btservice would adapt it in the future

huangyulong3 commented on service/stacks/zephyr/sal_adapter_interface.c:
```diff
@@ -399,10 +399,11 @@ static void zblue_on_bond_deleted(uint8_t id, const bt_addr_le_t* peer)
 }
 
-static void zblue_on_ready_cb(int err)
+static void zblue_on_ready_cb(uint8_t dev_id, int err)
```
Why doesn't used `controller_id_t id` here?

chengkai15 replied:
> Why doesn't used `controller_id_t id` here?

done

chejinxian reviewed (approved):
LGTM.
